### PR TITLE
Validations and tests

### PR DESCRIPTION
**Pull request by blueberry1312**
## In this PR, I did the following changes:

- Added validation to User and Post models
- Add tests files for User, Post, Like and Comment models.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      @user.posts_counter = 'a'
+      expect(@user).to_not be_valid
+    end
+
+    it 'should validate if posts_counter is greater than or equal to zero' do
+      expect(@user).to be_valid
+      @user.posts_counter = -1
+      expect(@user).to_not be_valid
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/models/user_spec.rb:40]: - [x] As required for the project to have a corresponding unit test to any custom methods in the model, kindly ensure that you also add test(s) to the `recent_posts` method as right now I cannot see their tests

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    end
+
+    it 'should update posts_counter of the user' do
+      user_posts_counter = @user.posts_counter
+      expect(user_posts_counter).to eq(0)
+
+      @user.reload
+      user_posts_counter = @user.posts_counter
+      expect(user_posts_counter).to eq(1)
+    end
```
Comment (tresorsawasawa) [inline spec/models/post_spec.rb:89]: - [x] As required for the project to have a corresponding unit test to any custom methods in the model, kindly ensure that you also add test(s) to the `update_post_counter` method  as right now I cannot see their tests

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    it 'should be a Like' do
+      expect(@like).to be_a(Like)
+    end
+
+    it 'should have the attributes' do
+      expect(@like).to have_attributes(post_id: @post.id, author_id: @user.id)
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/models/like_spec.rb]: - [x] As required for the project to have a corresponding unit test to any custom methods in the model, kindly ensure that you also add test(s) to the `update_like_counter` method as right now I cannot see their tests

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    it 'should be a Comment' do
+      expect(@comment).to be_a(Comment)
+    end
+
+    it 'should have the attributes' do
+      expect(@comment).to have_attributes(author_id: @user.id, post_id: @post.id,
+                                          text: 'This is my first comment')
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/models/comment_spec.rb]: - [x] As required for the project to have a corresponding unit test to any custom methods in the model, kindly ensure that you also add test(s) to the `update_comment_counter` method as right now I cannot see their tests

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @blueberry1312 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 2 above)
Comment (blueberry1312) [reply spec/models/post_spec.rb:89]: this method is written from the line 82-89

**Review by Munsa1 — APPROVED** [review]
Hi @blueberry1312, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congratulations-1

## Highlights
Made Requested Changes As Requested By The Previous Reviewer :heavy_check_mark: 
Good Communication :heavy_check_mark: 
Imperative Tense Used For Commit Messages :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**